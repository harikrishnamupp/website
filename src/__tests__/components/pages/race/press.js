import React from 'react'
import renderer from 'react-test-renderer'
import { useStaticQuery } from 'gatsby'
import Press from '~components/pages/race/press'

beforeEach(() => {
  useStaticQuery.mockImplementation(() => ({
    allContentfulRaceProjectNewsArticle: {
      nodes: [
        {
          publication: 'The Post',
          id: '10009',
          title: 'A race-related covid-19 article',
          publishDate: 'May 4, 2020',
          url: 'https://crouton.net',
        },
        {
          publication: 'The Journal',
          id: '12091',
          title: 'covid-19 title',
          publishDate: 'March 15, 2020',
          url: 'https://the-journal.org',
        },
        {
          publication: 'The Press',
          id: '12098',
          title: 'covid-19 race headline',
          publishDate: 'April 21, 2020',
          url: 'https://www.newsorg.com/2020/04/21/covid-19-race-headline/',
        },
      ],
    },
  }))
})

describe('Components : Race : Press', () => {
  it('renders correctly', () => {
    const tree = renderer.create(<Press />).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
