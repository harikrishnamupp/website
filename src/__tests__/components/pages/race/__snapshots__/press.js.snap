// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components : Race : Press renders correctly 1`] = `
<ul
  className="press-list pressList"
>
  <li>
    <cite
      className="title"
    >
      <a
        href="https://crouton.net"
      >
        A race-related covid-19 article
      </a>
    </cite>
    <div
      className="detail-text detailText"
    >
      <span
        className="source"
      >
        The Post
      </span>
      <span
        aria-hidden={true}
        className="dotSeparator"
      >
        •
      </span>
      May 4, 2020
    </div>
  </li>
  <li>
    <cite
      className="title"
    >
      <a
        href="https://the-journal.org"
      >
        covid-19 title
      </a>
    </cite>
    <div
      className="detail-text detailText"
    >
      <span
        className="source"
      >
        The Journal
      </span>
      <span
        aria-hidden={true}
        className="dotSeparator"
      >
        •
      </span>
      March 15, 2020
    </div>
  </li>
  <li>
    <cite
      className="title"
    >
      <a
        href="https://www.newsorg.com/2020/04/21/covid-19-race-headline/"
      >
        covid-19 race headline
      </a>
    </cite>
    <div
      className="detail-text detailText"
    >
      <span
        className="source"
      >
        The Press
      </span>
      <span
        aria-hidden={true}
        className="dotSeparator"
      >
        •
      </span>
      April 21, 2020
    </div>
  </li>
</ul>
`;
